editor.planx.uk/src/@planx/components/ContactInput/Public.tsx

```diff
@@ -38,14 +29,26 @@ export default function ContactInputComponent(props: Props): FCReturn {
       email: "",
     },
     onSubmit: (values) => {
-      // map internal props to expected granular passport keys before submitting
-      values["name.first"] = values.firstName;
-      values["name.last"] = values.lastName;
-      values["company.name"] = values.organisation;
-      values["phone.primary"] = values.phone;
+      // map values to the existing/expected passport structure to minimize conditional handling later in Send schemas, etc
+      const newPassportData: any = {};
+      newPassportData[`${props.fn}.title`] = values.title;
+      newPassportData[`${props.fn}.name.first`] = values.firstName;
+      newPassportData[`${props.fn}.name.last`] = values.lastName;
+      newPassportData[`${props.fn}.company.name`] = values.organisation;
+      newPassportData[`${props.fn}.phone.primary`] = values.phone;
+      newPassportData[`${props.fn}.email`] = values.email;
```

Comment:
This is a solid solution I think and you can see from this commit and it's doing a great job of stripping complexity out elsewhere 🎉